research/2022-07-15-relay-anonymity.md

```diff
@@ -0,0 +1,351 @@
+For our privacy and anonymity analysis, we start with the relay protocol because it is at the core of Waku v2 enabling Waku's publish subscribe approach to P2P messaging.
+In its current form, Waku relay is a minor extension of [libp2p GossipSub](https://github.com/libp2p/specs/blob/master/pubsub/gossipsub/README.md).
+
+## Informal Definitions: Security, Privacy, and Anonymity
+
+The concepts of security, privacy, and anonymity are linked and have quite a bit of overlap.
+
+### Security
+
+Of the three, [Security](https://en.wikipedia.org/wiki/Information_security) has the clearest agreed upon definition,
+at least regarding its key concepts: *confidentiality*, *integrity*, and *availability*.
+
+* confidentiality:  data is not disclosed to unauthorized entities.
+* integrity: data is not modified by unauthorized entities.
+* availability: data is available, i.e. accessible by authorized entities.
+
+While these are the key concepts, the definition of information security has been extended over time including further concepts,
+e.g. [authentication](https://en.wikipedia.org/wiki/Authentication) and [non-repudiation](https://en.wikipedia.org/wiki/Non-repudiation).
+We might cover these in future posts.
+
+### Privacy
+
+Privacy allows users to choose which data and information
+
+* they want to share
+* and with whom they want to share it.
+
+This includes data and information that is associated with and/or generated by users.
+Protected data also comprises metadata that might be generated without users being aware of.
+This means, no further information about the sender or the message is leaked.
+Metadata that is protected as part of the privacy-preserving property does not cover protecting the identities of sender and receiver.
+Identities are protected by the [anonymity property](#anonymity).
+
+Often privacy is realized by the confidentiality property of security.
+This neither makes privacy and security the same, nor the one a sub category of the other.
+While security is abstract itself (its properties can be realized in various ways), privacy lives on a more abstract level using security properties.
+Privacy typically does not use integrity and confidentiality. An adversary who has no access to the private data, because the message has been encrypted, could still alter the message.
+
+Waku offers confidentiality, because message content is only disclosed to the indented receivers.
+We have various mechanisms in place to guarantee confidentiality.
+The strongest option are secure channels setup with the help of the [Noise Protocol Framework](https://noiseprotocol.org/).
+These secure channels also provide good metadata protection properties.
+However, we do not have a metadata protection analysis as of yet,
+which is part of our privacy/anonymity roadmap.
+Only after completing this analysis, and after addressing possibly upcoming issues, we will declare Waku v2 as privacy-preserving with respect to metadata protection.
```

Comment:
we de-facto already are declaring it this though, so this seems a bit contradictory

Reply:
I think we have done some things to mitigate timestamp leak when it comes to time resolution, cc @jm-clius

Reply:
Our plan is to have sender timestamps "tick" along coarse time resolution boundaries (say only every 500ms) which would decrease the likelihood of correlation attacks (as many messages will fall in the same time bucket). This is not implemented yet and will have an effect on i.a. the sorting order of messages in different store nodes. Not sure if this fits into this article that treats the message itself as a blackbox, though may be worth mentioning as a future plan.

Reply:
(Of course, nothing prevents senders from already using coarse timestamp resolutions, which we could add as an implementation suggestion elsewhere.)